src/main/java/core/basesyntax/ShopServiceImpl.java

```diff
@@ -0,0 +1,19 @@
+package core.basesyntax;
+
+import java.util.List;
+
+public class ShopServiceImpl implements ShopService {
+    private final OperationStrategy operationStrategy;
+
+    public ShopServiceImpl(OperationStrategy operationStrategy) {
+        this.operationStrategy = operationStrategy;
+    }
+
+    @Override
+    public void process(List<FruitTransaction> transactions, Storage storage) {
+        for (FruitTransaction transaction : transactions) {
+            OperationHandler handler = operationStrategy.getHandler(transaction.getOperation());
+            handler.handle(transaction, storage);
+        }
+    }

```

Comment:
The `process` method should not directly depend on the `Storage` class. Instead, consider using dependency injection to pass the necessary storage object to the `OperationHandler` implementations. This will make your service more independent and adhere to SOLID principles.